distod/src/main/scala/com/github/codelionx/distod/actors/partitionMgmt/PartitionManager.scala

```diff
@@ -250,31 +246,33 @@ class PartitionManager(
     }
   }
 
-  private def generateStrippedPartitionJobs(
-      key: CandidateSet,
-      pendingJobs: PendingJobMap[CandidateSet, PendingResponse],
-      pendingResponse: PendingResponse
-  ): PendingJobMap[CandidateSet, PendingResponse] = {
+  private def generateStrippedPartitionJobs(key: CandidateSet, pendingResponse: PendingResponse): Unit = {
     timings.time("Partition generation") {
       generatorPool match {
         case Some(pool) =>
-          val jobs = JobChainer.calcJobChain(key, partitions)
-          if (jobs.size > 20) {
-            context.log.warn("Generating expensive job chain of size {}: {}", jobs.size, jobs.map(_.key.size).sorted)
+          lazy val jobs = JobChainer.calcJobChain(key, partitions)
+          if (settings.directPartitionProductThreshold <= 1) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Consider using case matching instead of else if blocks](https://app.codacy.com/manual/CodeLionX/distod/pullRequest?prid=5048550)